<launch>    
  <!-- 地图尺寸参数 -->
  <arg name="map_size_x" default="42.0"/>
  <arg name="map_size_y" default="30.0"/>
  <arg name="map_size_z" default="5.0"/>

  <!-- 其他参数 -->
  <arg name="odom_topic" default="visual_slam/odom"/>

  <!-- map generator -->
  <node pkg="map_generator" name="random_forest" type="random_forest" output="screen">    
    <param name="map/x_size"     value="36"/>
    <param name="map/y_size"     value="20"/>
    <param name="map/z_size"     value="3"/>
    <param name="map/resolution" value="0.1"/>        

    <param name="ObstacleShape/seed" value="1"/>

    <param name="map/obs_num"    value="200"/>
    <param name="ObstacleShape/lower_rad" value="0.5"/>
    <param name="ObstacleShape/upper_rad" value="0.7"/>
    <param name="ObstacleShape/lower_hei" value="0.0"/>
    <param name="ObstacleShape/upper_hei" value="3.0"/>        

    <param name="map/circle_num" value="200"/>        
    <param name="ObstacleShape/radius_l" value="0.7"/>        
    <param name="ObstacleShape/radius_h" value="0.5"/>        
    <param name="ObstacleShape/z_l" value="0.7"/>        
    <param name="ObstacleShape/z_h" value="0.8"/>        
    <param name="ObstacleShape/theta" value="1"/>        

    <param name="sensing/radius" value="5.0"/>        
    <param name="sensing/rate"   value="1.0"/>  

    <param name="min_distance" value="1.2"/>         
  </node>

</launch>
